lib/charms/postgresql_k8s/v0/postgresql_tls.py

```diff
@@ -0,0 +1,181 @@
+    generate_csr,
+    generate_private_key,
+)
+from cryptography import x509
+from cryptography.x509.extensions import ExtensionType
+from ops.charm import ActionEvent
+from ops.framework import Object
+from ops.pebble import PathError, ProtocolError
+
+# The unique Charmhub library identifier, never change it
+LIBID = "c27af44a92df4ef38d7ae06418b2800f"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version.
+LIBPATCH = 1
+
+logger = logging.getLogger(__name__)
+SCOPE = "unit"
+TLS_RELATION = "certificates"
+
+
+class PostgreSQLTLS(Object):
+    """In this class we manage certificates relation."""
+
+    def __init__(self, charm, peer_relation):
+        """Manager of PostgreSQL relation with TLS Certificates Operator."""
+        super().__init__(charm, "client-relations")
+        self.charm = charm
+        self.peer_relation = peer_relation
+        self.certs = TLSCertificatesRequiresV1(self.charm, TLS_RELATION)
+        self.framework.observe(
+            self.charm.on.set_tls_private_key_action, self._on_set_tls_private_key
+        )
+        self.framework.observe(
+            self.charm.on[TLS_RELATION].relation_joined, self._on_tls_relation_joined
+        )
+        self.framework.observe(
+            self.charm.on[TLS_RELATION].relation_broken, self._on_tls_relation_broken
+        )
+        self.framework.observe(self.certs.on.certificate_available, self._on_certificate_available)
+        self.framework.observe(self.certs.on.certificate_expiring, self._on_certificate_expiring)
+
+    def _on_set_tls_private_key(self, event: ActionEvent) -> None:
+        """Set the TLS private key, which will be used for requesting the certificate."""
+        self._request_certificate(event.params.get("private-key", None))
+
+    def _request_certificate(self, param: Optional[str]):
+        """Request a certificate to TLS Certificates Operator."""
+        if param is None:
+            key = generate_private_key()
+        else:
+            key = self._parse_tls_file(param)
+
+        csr = generate_csr(
+            private_key=key,
+            subject=self.charm.get_hostname_by_unit(self.charm.unit.name),
+            sans=self._get_sans(),
+            additional_critical_extensions=self._get_tls_extensions(),
+        )
+
+        self.charm.set_secret(SCOPE, "key", key.decode("utf-8"))
+        self.charm.set_secret(SCOPE, "csr", csr.decode("utf-8"))
+
+        if self.charm.model.get_relation(TLS_RELATION):
+            self.certs.request_certificate_creation(certificate_signing_request=csr)
+
+    @staticmethod
+    def _parse_tls_file(raw_content: str) -> bytes:
+        """Parse TLS files from both plain text or base64 format."""
+        if re.match(r"(-+(BEGIN|END) [A-Z ]+-+)", raw_content):
+            return re.sub(
+                r"(-+(BEGIN|END) [A-Z ]+-+)",
```

Comment:
Making this a variable would make it clearer what you're matching for here.

Reply:
Update on https://github.com/canonical/postgresql-k8s-operator/pull/35/commits/74c09adc07395ab4d276719b67cbe8cdf578e393.